### ISSUE TITLE
Validator gives an error for header element

### ISSUE THREAD

**Issue by programmer3000** [w3c/html#1221]
If you go to link https://www.w3.org/TR/html/sections.html#the-header-element  and watch Example 20 and then if you go to validator https://validator.w3.org/  and will check that example  it will return error: "The element header must not appear as a descendant of the header element." though example 20 according HTML 5.2 full clear.
Although HTML 5.2 let be content model where header could be placed inside header (but if insider header inside sectioning content) "Content model: Flow content, but with no main element descendants, or header, footer elements that are not descendants of sectioning content which is a descendant of the header. " - I think it's very bad idea if `<header>` can placed incide `<header>` (same for `<footer>`). Because `<header>` element represents introductory content. And if document will have two `<header>`, document will more unreadable with strange semantic, 
For example - if remove one header - it will perfect for semantic, logic and for reading: (as author think what inside `<header>` - represents introductory content for `<article>`)
```
<article>
  <header>
    <h1>Flexbox: The definitive guide</h1>
    <aside>
        <h2>About the author: Wes McSilly</h2>
        <p><a href="./wes-mcsilly/">Contact him! (Why would you?)</a></p>
      <p>Expert in nothing but Flexbox. Talented circus sideshow.</p>
    </aside>
  </header>
  <p><ins>The guide about Flexbox was supposed to be here, but it
    turned out Wes wasn’t a Flexbox expert either.</ins></p>
</article>
```
So for clear that mistake most right go to link https://www.w3.org/TR/html/sections.html#the-header-element and rewrite content model to; 
Must be like this: "Content model:
    Flow content, but with no main, header or footer elements descendants"

**Comment by chaals** [comment]
@programmer3000 this structure was allowed - although it is probably usually a bad idea - in resolving #273. I think the use case that makes the most sense is when mixing components together, so a component which contains a `header` is OK to be inside a `header`.

Unless we want to revert that change, this should be a bug on the validator rather than the spec.

ping @stevefaulkner

**Comment by programmer3000** [comment]
@chaals  Yes, now it's bug of validator. But main problem in the structure.
It very bad idea let be ok header inside header. So possible let everytag be everywhere and what will be in the end? - different understanding of the structure by different developers. Such situation now - too much topics about structure - because structure is not explicitly.

**Comment by programmer3000** [comment]
@chaals and so what will do with that problem w3c?

**Comment by chaals** [comment]
@programmer3000 we will look at the issue. If something matches a real-world need, and works, unless we have a better solution and a clear explanation of how to change what people do, we will probably make it valid in the spec.

We might add advice saying "don't do this if you don't have to", and explanations of why. It is probably a good idea to add such explanations in the case of `header` inside a `header`.

**Comment by chaals** [comment]
I suggest the resolution of this should be to explain how a `header` inside a `header` might appear (e.g. by including components made separately in a template), and why it is probably not a good idea. That would be editorial and we can probably get it done in the current milestone.

**Comment by programmer3000** [comment]
@chaals  I think it no good idea "header inside a header might appear (e.g. by including components made separately in a template)" - so can everytag be inside everytag "(e.g. by including components made separately in a template)". 
I think it must be phrase like: "Possible, but not recommend use `<header>` inside `<header>` even if "descendant, because it born implicit structure" - same for `<footer>`
